plugins/debug-console.js

```diff
@@ -18,41 +18,46 @@ var debugTab = {};
 // DEBUGGING TOOLS ///////////////////////////////////////////////////
 // meant to be used from browser debugger tools and the like.
 
-
-debugTab.renderDetails = function() {
-  debugTab.console.log('portals: ' + Object.keys(window.portals).length);
-  debugTab.console.log('links:   ' + Object.keys(window.links).length);
-  debugTab.console.log('fields:  ' + Object.keys(window.fields).length);
-};
-
-debugTab.printStackTrace = function() {
-  var e = new Error('dummy');
-  debugTab.console.error(e.stack);
-  return e.stack;
-};
-
-debugTab.console = {};
-debugTab.console.show = function() {
-  $('#chat, #chatinput').show();
-  $('#chatinput mark').css('cssText', 'color: #bbb !important').text('debug:');
-  $('#chat > div').hide();
-  $('#debugconsole').show();
-  $('#chatcontrols .active').removeClass('active');
-  $("#chatcontrols a:contains('debug')").addClass('active');
+debugTab.create = function () {
+  window.chat.addChannel({
+    id: 'debug',
+    name: 'Debug',
+    inputPrompt: 'debug:',
+    inputClass: 'debug',
+    sendMessage: function (_, msg) {
+      var result;
+      try {
+        result = eval(msg);
+      } catch (e) {
+        if (e.stack) {
+          console.error(e.stack);
+        }
+        throw e; // to trigger native error message
+      }
+      if (result !== undefined) {
+        console.error(result.toString());
+      }
+      return result;
+    },
+  });
 };
 
-debugTab.console.renderLine = function(errorType, args) {
+debugTab.renderLine = function (errorType, args) {
   args = Array.prototype.slice.call(args);
   var color = '#eee';
   switch (errorType) {
-  case 'error':   color = '#FF424D'; break;
-  case 'warning': color = '#FFDE42'; break;
+    case 'error':
+      color = '#FF424D';
```

Comment:
CSS classes instead of colors? And then colors in css? 🤷

Reply:
yes, definitely